dace/transformation/dataflow/map_distribution.py

```diff
@@ -0,0 +1,940 @@
+                if list(desc.shape) != m.subset.size_exact():
+                    # Second attempt
+                    # TODO: We need a solution for symbols not matching
+                    if str(list(desc.shape)) != str(m.subset.size_exact()):
+                        raise NotImplementedError
+            outputs.add(dst)
+
+        for out in outputs:
+            desc = out.desc(sdfg)
+            if isinstance(desc, data.Scalar):
+                raise NotImplementedError
+            if isinstance(desc, data.Array):
+                local_name, local_arr = sdfg.add_temp_transient(
+                    [(desc.shape[0]) // Px, (desc.shape[1]) // Py],
+                    dtype=desc.dtype, storage=desc.storage)
+                local_access = graph.add_access(local_name)
+                bsizes_name, bsizes_arr = sdfg.add_temp_transient(
+                    (2,), dtype=dace.int32)
+                bsizes_access = graph.add_access(bsizes_name)
+                bsizes_tasklet = nodes.Tasklet(
+                    '_set_bsizes_',
+                    {}, {'__out'},
+                    "__out[0] = {x}; __out[1] = {y}".format(
+                        x=(desc.shape[0]) // Px, y=(desc.shape[1]) // Py))
+                graph.add_edge(bsizes_tasklet, '__out', bsizes_access, None,
+                               dace.Memlet.from_array(bsizes_name, bsizes_arr))
+                scatter_node = BlockCyclicGather('_Gather_')
+                graph.add_edge(local_access, None, scatter_node, '_inbuffer',
+                               dace.Memlet.from_array(local_name, local_arr))
+                graph.add_edge(bsizes_access, None, scatter_node, '_block_sizes',
+                               dace.Memlet.from_array(bsizes_name, bsizes_arr))
+                graph.add_edge(scatter_node, '_outbuffer', out, None,
+                               dace.Memlet.from_array(out.data, desc))
+
+                for e in graph.edges_between(map_exit, out):
+                    graph.add_edge(map_exit, e.src_conn, local_access, None,
+                                   dace.Memlet.from_array(local_name, local_arr))
+                    graph.remove_edge(e)
+                for e in graph.in_edges(map_exit):
+                    if e.data.data == out.data:
+                        e.data.data = local_name
+                continue
+            raise NotImplementedError
+
+        map_entry.map.params = params
+        map_entry.map.range = subsets.Range(ranges)
+
+
+@registry.autoregister_params(singlestate=True, strict=False)
+class RedundantComm2D(pm.Transformation):
+    """ Implements the redundant array removal transformation, applied
+        when a transient array is copied to and from (to another array),
+        but never used anywhere else. """
+
+    in_array = pm.PatternNode(nodes.AccessNode)
+    gather = pm.PatternNode(nodes.Tasklet)
+    mid_array = pm.PatternNode(nodes.AccessNode)
+    scatter = pm.PatternNode(nodes.Tasklet)
```

Comment:
These should be the specific library nodes.

Reply:
It does not work in general. The issue is that the pattern might emerge only after expanding the library nodes, at which point you have tasklets and not library nodes anymore. You can lookup the commit history of the file where I have commented out an unsuccessful attempt at matching the library nodes.